### PR TITLE
Update smirnoff99Frosst to remove generics in DrugBank

### PR DESCRIPTION
**Pull request by bannanc**
This pull request will include parameters needed to match most of the chemistry in the larger set DrugBank. It still is no metals and molecules with <= 200 heavy atoms.

In this message I will list all of the changes I've made with links to issues where they have been discussed. 

### Parm99 or Parm@Frosst

**Bonds**
* Remove neutral requirement in the bonds to hypervalent sulfur ( `[#16X3+0,#16X4]`) (issue #216 )
    - The relevant atom type from parm@Frosst is applied to at least neutrally and negatively charged trivalent sulfurs. So the plan was to remove the +0
    - There is a special case for sulfur carbon bonds. bonds for divalent (not hypervalent) sulfur come before these hypervalent bonds. In that section there is a S-C bond for `[#16X2,#16X1-1,#16X3+1:1]` sulfurs. We do not want the hypervalent sulfur bond to over ride these S-C bonds. So the SMIRKS pattern for hypervalent sulfur singly bound to carbon was set as `[#16X4,#16X3!+1:1]-[#6]`
* Add C=S bond from parm@frosst (issue #216 )
* removed `X1` requirement for nitrogen in nitrile (issue #216 )
    - the atomtype in parm99/parm@frosst matches `#7X1` and `#7X2&(*#*)` so it is the triple bond that is important. 
* removed `X4` requirement for the most generic S-C bond. (issue #216 )
* C-[halogen] bonds to cover all carbons ([smirnoff99Frosst issue#27](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/27))
* remove `X2` requirement from sulfur in S-H bond ([smirnoff99Frosst issue#31](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/31))
* Made some bonds more generic to address odd corner cases ([smirnoff99Frosst issue#34](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/34))
    - Remove X3 requirements for most generic N-N to cover N+1 cases
    - Change `[#6X4:1]-[#8X2:2] ` to `[#6:1]-[#8:2]` to cover corner cases
    - Remove X2 on carbon in nitrile bonds to address charged corner cases
* Changed order of carbonyl bonds to properly type carbonyl groups ([smirnoff99Frosst issue#38](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/38))
* Changed the first carbonyl group to be more general for groups of the type X=C=O as discussed in issue [smirnoff99Frosst issue#37](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/37)

**Angles**
* added slightly more generic `#7X2` angle for linear case (R-C#N+1-R)
* made the X-N-O angles in nitro groups generic enough for X to be carbon, oxygen, or nitrogen ([smirnoff99Frosst issue#36](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/36))
* Updated most generic divalent sulfur to accomodate aromatic cases ([smirnoff99Frosst issue#34](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/34))
    - **waiting approval**

**Torsions**
* Updated aromatic nitrogen torsion to cover all generic -N:C- cases ([smirnoff99Frosst issue#39](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/39))
    - **waiting for approval**
* Remove neutral requirement in the bonds to hypervalent sulfur ( `[#16X3+0,#16X4]`) ([smirnoff99Frosst issue#40](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/40) )
    - The relevant atom type in parm@Frosst is SO which is for all `#16X3`, but in smirnoff there are some higher priority cases with `#16X3+1` we wanted to remove the neutral requirement without overwriting those. 
    - **waiting approval**
* Added torsion for nitro group-tetrahedral carbon based on parm@Frosst and GAFF2 ([smirnoff99Frosst issue#42](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/42))
    - **waiting approval**

### Estimates from GAFF2
**Bonds**
* S-P bond ([smirnoff99Frosst issue#20](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/20))
* H-P bond ([smirnoff99Frosst issue#21](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/21))
* N-halogen bonds ([smirnoff99Frosst issue#24](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/24))
* P-halogen bonds ([smirnoff99Frosst issue#26](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/26))
* S-halogen bonds ([smirnoff99Frosst issue#33](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/33))

**Angles**

**Torsions**
* -O-O- torsion ([smirnoff99Frosst issue#23](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/23))
* -S=,:-C- torsion ([smirnoff99Frosst issue#41](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/41))
    - **waiting approval**
* -P-N- torsions ([smirnoff99Frosst issue#32](https://github.com/open-forcefield-group/smirnoff99Frosst/issues/32))
    - **waiting approval**

### CODE REVIEW

**Comment by davidlmobley** [comment]
@bannanc - I'm going to suggest edits to your summary above since this is our record for posterity. Here are a couple of things which are unclear to me:

> Our original plan was to remove the +0, however, there were bonds specified above for the case of `#16+1` to carbon so to avoid overwriting those, the carbon bond was changed to `[#16X4,#16X3!+1]`

What's "specified above" mean? I'm guessing you mean "above this in the force field file"? Also, the reference of "the carbon bond" could be more clear -- THIS carbon bond? Or the one above? (Again, presumably you mean the former, but it's easier if it's just clear when you write it the first time so one doesn't have to stop and think about it.)

**Comment by bannanc** [comment]
@davidlmobley 
Assuming you approve the newest round of changes. This now types all the filtered drugbank molecules, that have tripos mol2 files. 

Note - there are a few corner cases we don't quite catch in the full DrugBank set. 

My next plan is to close all the issues this pull request addresses and then open a single (LONG) issue with chemistry that we think needs to be explored further. That will include anything we got from GAFF2, things we don't cover now (i.e. boron/silicon), and anything where we made things fairly generic to cover our bases. This will serve as a source of molecules/functional groups for the undergrad in our group during the summer.

**Comment by bannanc** [comment]
For reference:

[Charged Tripos mol2 files of filtered drugBank](https://drive.google.com/drive/folders/0BwF2-3puCvfETVQ3SzN4WkE5bDQ)

[GAFF2 typed molecules for filtered DrugBank](https://drive.google.com/drive/folders/0BwF2-3puCvfEcldJTE8xZHVmdEU)

**Comment by davidlmobley** [comment]
@bannanc :

> My next plan is to close all the issues this pull request addresses and then open a single (LONG) issue with chemistry that we think needs to be explored further. That will include anything we got from GAFF2, things we don't cover now (i.e. boron/silicon), and anything where we made things fairly generic to cover our bases. This will serve as a source of molecules/functional groups for the undergrad in our group during the summer.

This plan sounds excellent.

Reviewing FF changes in a cursory way (since I already reviewed all the individual issues in detail) now.

**Comment by davidlmobley** [comment]
And, once merged, you will also migrate these changes to github.com/open-forcefield-group/openforcefield and github.com/open-forcefield-group/smirnoff99frosst, @bannanc ? (We're working towards removing smirnoff from smarty).

File: utilities/convert_frosst/smirffishFrcmod.parm99Frosst.txt
Comment (davidlmobley) [inline utilities/convert_frosst/smirffishFrcmod.parm99Frosst.txt]: "ethers" not "eithers". :)

**Review by davidlmobley — APPROVED** [review]
LGTM, @bannanc !

**Comment by bannanc** [comment]
@davidlmobley if you're ok with what I did with the aromatic nitrogen torsion then I think this is good to know with the knowledge that we'll probably have another one in the not too distant future.

**Comment by davidlmobley** [comment]
@bannanc - tests are failing because the tests load `smirff99Frosst.ffxml`  but you renamed to `smirnoff99Frosst.ffxml`. Can you fix?

> `Failure: ValueError (Sorry! /home/travis/miniconda/envs/test/lib/python2.7/site-packages/smarty/data/forcefield/smirff99Frosst.ffxml does not exist. If you just added it, you'll have to re-install) ... ERROR`

**Comment by davidlmobley** [comment]
(It's OK with me if you go ahead and merge once tests pass, @bannanc !)

**Comment by davidlmobley** [comment]
@bannanc - looks like py27 and py35 passed on your last commit, but py34 is failing. Checking.

**Comment by bannanc** [comment]
The most recent run failed on python3 and not python2, but the failure said it didn't get 100% on the AlkEthOH answers... I don't think we changed anything in the AlkEthOH parameters on this set. I'll see if the same problem repeats on this test

**Comment by davidlmobley** [comment]
Well, that's really weird... Py34 test failed with something you haven't touched in here:

```
FAIL: Test score is 100% if correct VdW, Bond, Angles, or Torsions
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/lib/python3.4/site-packages/smarty/tests/test_smirky_sampler.py", line 53, in test_correct_fragments
    self.assertAlmostEqual(fracfound, 1.0, msg = "Not finding 100%% of AlkEthOH when starting from correct %s SMIRKS." % typetag)
AssertionError: 0.9962640099626401 != 1.0 within 7 places : Not finding 100% of AlkEthOH when starting from correct VdW SMIRKS.
```

I think the SMIRKY test may have accepted a very rare move which resulted in losing some coverage of the AlkEthOH set, perhaps, just in this one case? Looks like it should be good to merge when you are ready, @bannanc .

**Comment by bannanc** [comment]
~I'm going to make the number of steps on the 100% test smaller to avoid that happening in the future.~ 

Nevermind, it already has 2 steps and temperature = 0.0 it shouldn't ever accept moves that decrease the score... I don't know what caused the failed test assuming this one passes I'll merge it.

**Pull request merged**
